src/app/owners/owner-add/owner-add.component.html

```diff
@@ -18,65 +18,72 @@
 
 
 
-<div class="container-fluid">
-  <div class="container xd-container">
-    <h2>
-      New Owner
-    </h2>
-    <form (ngSubmit)="onSubmit(ownerForm.value)" #ownerForm="ngForm" class="form-horizontal">
-      <div class="form-group" hidden="true">
-        <input type="text" hidden="true" class="form-control" id="id" [(ngModel)]="owner.id" name="id"/>
-      </div>
-      <div class="form-group has-feedback" [class.has-success]="firstName.dirty && firstName.valid" [class.has-error]="firstName.dirty &&  !firstName.valid">
-        <label for="firstName" class="col-sm-2 control-label">First Name</label>
-        <div class="col-sm-10">
-          <input type="text" class="form-control" id="firstName" [(ngModel)]="owner.firstName" minlength="2" required name="firstName" #firstName="ngModel"/>
-          <span class="glyphicon form-control-feedback" [class.glyphicon-ok]="firstName.valid" [class.glyphicon-remove]="!firstName.valid" aria-hidden="true"></span>
-          <span class="help-block" *ngIf="firstName.dirty && firstName.hasError('required')">First name is required</span>
-          <span class="help-block" *ngIf="firstName.dirty && firstName.hasError('minlength')">First name must be at least 2 characters long</span>
+  <div class="container-fluid">
+    <div class="container xd-container">
+      <h2>
+        New Owner
+      </h2>
+      <form (ngSubmit)="onSubmit(ownerForm.value)" #ownerForm="ngForm" class="form-horizontal">
+        <div class="form-group" hidden="true">
+          <input type="text" hidden="true" class="form-control" id="id" [(ngModel)]="owner.id" name="id"/>
         </div>
-      </div>
-      <div class="form-group has-feedback" [class.has-success]="lastName.dirty && lastName.valid" [class.has-error]="lastName.dirty && !lastName.valid">
-        <label for="lastName" class="col-sm-2 control-label">Last Name</label>
-        <div class="col-sm-10">
-          <input type="text" class="form-control" id="lastName" [(ngModel)]="owner.lastName" name="lastName" minlength="2" required #lastName="ngModel"/>
-          <span class="glyphicon form-control-feedback" [class.glyphicon-ok]="lastName.valid" [class.glyphicon-remove]="!lastName.valid" aria-hidden="true"></span>
-          <span class="help-block" *ngIf="lastName.dirty && lastName.hasError('required')">Last name is required</span>
-          <span class="help-block" *ngIf="lastName.dirty && lastName.hasError('minlength')">Last name must be at least 2 characters long</span>
+        <div class="form-group has-feedback" [class.has-success]="firstName.dirty && firstName.valid" [class.has-error]="firstName.dirty &&  !firstName.valid">
+          <label for="firstName" class="col-sm-2 control-label">First Name</label>
+          <div class="col-sm-10">
+            <input type="text" class="form-control" id="firstName" [(ngModel)]="owner.firstName" minlength="1" maxlength="30" pattern="^[a-zA-Z]*$" required name="firstName" #firstName="ngModel"/>
```

Comment:
For `lastname`and `firstname` I think we could authorize `space` and `dash` characters in the frontend and the backend. We could have compound name like `Jean-Baptiste`. 
What you think about @alexandre-touret ?

Reply:
This should happen with pet owner and veterinarien names. The definition of a pattern is not trivial. 

D‘Artagnon - quotes
Peterson Jr.  dot and inside single space
Jean- Baptiste inside dash


A possible solution is :

^[a-zA-Z]+([ \-']{0,1}[a-zA-Z]+){0,2}[.]{0,1}$

taken from 

https://a-tokyo.medium.com/first-and-last-name-validation-for-forms-and-databases-d3edf29ad29d

Reply:
Yes it's a good compromise. We don't support extended ASCII character for other fields.

Reply:
We could create another issue for this enhancement.